src/System.Private.CoreLib/shared/System/IO/Stream.cs

```diff
@@ -1327,6 +1298,9 @@ public override void WriteByte(byte b)
 
             public override IAsyncResult BeginWrite(byte[] buffer, int offset, int count, AsyncCallback callback, Object state)
             {
+#if CORERT
+                throw new PlatformNotSupportedException(); // TODO: https://github.com/dotnet/corert/issues/3251
```

Comment:
In corert we have `NotImplementedException` (refer to [this](https://github.com/dotnet/corert/blob/ed0b32ee6b2c3c9d7c9e1c21ca019cd60278f861/src/System.Private.CoreLib/src/System/IO/Stream.cs#L781)). 

Do we want to switch behavior to `PlatformNotImplementedException`?